ee/packages/abac/src/index.ts

```diff
@@ -442,6 +442,38 @@ export class AbacService extends ServiceClass implements IAbacService {
 		}));
 	}
 
+	async checkUsernamesMatchAttributes(usernames: string[], attributes: IAbacAttributeDefinition[]): Promise<void> {
+		if (!usernames.length || !attributes.length) {
+			return;
+		}
+
+		const nonComplianceConditions = this.buildNonCompliantConditions(attributes);
+		const nonCompliantUsersFromList = await Users.find(
+			{
+				username: { $in: usernames },
+				$or: nonComplianceConditions,
+			},
+			{ projection: { username: 1 } },
+		)
+			.map((u) => u.username as string)
+			.toArray();
+
+		const nonCompliantSet = new Set<string>(nonCompliantUsersFromList);
+
+		if (nonCompliantSet.size) {
+			// Note: open to suggestions, or if it's actually needed. My idea is to return the list of non compliant users, but our current errors dont' allow that
+			// Maybe we should just throw a generic error? idk. I may create a custom error just for this
+			const err = new Error('error-usernames-not-matching-abac-attributes');
```

Comment:
Shouldn't we use `MeteorError` from `@rocket.chat/core-services` here?

Reply:
Shoot, i always forget we have this one on services and not on meteor only.